### PR TITLE
Add data field to Instantiate response

### LINKED ISSUE #385: Add MsgInstantiateContractResponse.data
[MsgExecuteContractResponse](https://github.com/CosmWasm/wasmd/blob/v0.14.1/x/wasm/internal/types/tx.proto#L80-L84) and [MsgMigrateContractResponse](https://github.com/CosmWasm/wasmd/blob/v0.14.1/x/wasm/internal/types/tx.proto#L98-L103) have a contract defined `data` field. [MsgInstantiateContractResponse](https://github.com/CosmWasm/wasmd/blob/v0.14.1/x/wasm/internal/types/tx.proto#L62-L66) does not have this for historic reasons (we returned the raw contract address before).

Now that we can easily return structured proto messages, it would be good to add `MsgInstantiateContractResponse.data` for consistency.

### PR DESCRIPTION
**Pull request by alpe**
Resolves #385 

New field in `MsgInstantiateContractResponse` and minor refactoring to testing framework.

### CODE REVIEW

File: x/wasm/internal/types/tx.proto [hunk 1]
```diff
@@ -63,6 +63,8 @@ message MsgInstantiateContract {
 message MsgInstantiateContractResponse {
   // Address is the bech32 address of the new contract instance.
   string address = 1;
+  // Data contains base64-encoded bytes to returned from the contract
+  bytes data = 3;
```
Comment (ethanfrey) [inline x/wasm/internal/types/tx.proto:67]: Why 3? and not 2?

**Review by ethanfrey — APPROVED** [review]
Looks good. One little nit.

I think you have a good point of the data not being returned for any of the calls via governance. I would address that in another PR, as that is not just about instantiate.

**Comment by alpe** [comment]
I have created https://github.com/CosmWasm/wasmd/issues/431 to follow up on the events

**Comment by alpe** [comment]
Sorry, had to push -f after merge conflicts.

**Comment by ethanfrey** [comment]
> I have created #431 to follow up on the events

Great. and for 0.17 :smile:  (Actually events are non-breaking, so that could be 0.16.1).

Feel free to merge when CI finally runs

**Pull request merged**